.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: ROS CI
+
+on: [push, pull_request]
+
+jobs:
+  build-native:
+    runs-on: ubuntu-20.04
+    container:
+      image: swarmus/embedded
+    steps:
+      - name: Create catkin_ws
+        run: |
+          mkdir -p catkin_ws/src
+          cd catkin_ws
+          . /opt/ros/$ROS_DISTRO/setup.sh
+          catkin_make
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          path: 'catkin_ws/src/SwarmUS-ROS'
+
+      - name: Checkout submodules
+        run: git submodule update --init --recursive
+        working-directory: 'catkin_ws/src/SwarmUS-ROS'
+
+      - name: Install ROS dependencies
+        run: |
+          rosdep update
+          apt update
+          rosdep install --from-paths src --ignore-src -r -y
+        working-directory: 'catkin_ws'
+
+      - name: Build
+        run: |
+          . ./devel/setup.sh
+          catkin_make
+        working-directory: 'catkin_ws'
+
+      - name: Unit Tests
+        run: |
+          . ./devel/setup.sh
+          catkin_make run_tests
+        working-directory: 'catkin_ws'
+      
+      - name: Check Format
+        run: |
+          python src/SwarmUS-ROS/tools/contrib/run-clang-format.py --recursive --style=file src/SwarmUS-ROS/src/
```

Comment:
At the beginning, I did something similar, but I had some conflicts with the targets' name between the HiveMind and the other packages. Also, doing that would force me to redefine a new custom target in each package (I think?), which seems redundant to me. The python script just works for all the code in the /src folder, and will work with every future package we will create, without any setup.

Reply:
You could probably make a root cmake file in your src, so you would have src/CMakeLists.txt, add the subdirectories, add your format calls there, etc. I mean it's fine with the python script this way too. I was more worried about the doc than the format but since we want package per package.